### PR TITLE
[CHART] Support additional taints tolerations

### PR DESCRIPTION
**Pull request by k0da**
## Problem
chart have no ability to tolerate custom taints leading to chart not installable on tainted nodes
 
## Solution
Introduce additional value to add tolerations
 
## Testing
run through `helm template -n cattle-system rancher ./  --debug -f foo.yaml --show-only templates/deployment.yaml`

with foo.yaml containing:
```yaml
tolerations:
- key: "node.cloudprovider.kubernetes.io/uninitialized"
  value: "true"
  effect: "NoSchedule"
```

### Automated Testing

* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

File: chart/templates/deployment.yaml
Comment (k0da) [reply chart/templates/deployment.yaml]: Thanks for review, done

File: chart/values.yaml
Comment (mallardduck) [inline chart/values.yaml:183]: Can we rename this to `additionalTolerations`?

After looking at this more I think that's better. Since it's more accurate to situation and there's nothing lost being more descriptive/specific here.

File: chart/templates/deployment.yaml
Comment (mallardduck) [inline chart/templates/deployment.yaml:67]: ack - These should be renamed to match the new values name.

**Review by mallardduck — CHANGES_REQUESTED** [review]
Thanks for making those changes from my initial feedback. After looking this over further there are a few other tweaks I'd like to see made. Once those are addressed this can be re-reviewed and I'll merge if it looks ok.

Also, if you can do so and want to please add a test for the deployment (to verify tolerations). Otherwise after the requested changes I can just amend the PR before merging to include them.

Thanks in advance

File: chart/values.yaml
Comment (k0da) [reply chart/values.yaml:183]: Make sense, are you ok with extraTolerations?

Comment (mallardduck) [reply chart/values.yaml:183]: Yeah, that works for me. Thanks

**Comment by k0da** [comment]
> Thanks for making those changes from my initial feedback. After looking this over further there are a few other tweaks I'd like to see made. Once those are addressed this can be re-reviewed and I'll merge if it looks ok.
> 
> Also, if you can do so and want to please add a test for the deployment (to verify tolerations). Otherwise after the requested changes I can just amend the PR before merging to include them.
> 
> Thanks in advance

Can you point me at where tests are defined?

**Comment by mallardduck** [comment]
Hiya, @k0da  For this instance you can add a new test here: https://github.com/rancher/rancher/blob/main/chart/tests/deployment_test.yaml

It could be a simple set of tests that checks the count of tolerations. One to expect a single toleration to cover the default, and another to cover adding the toleration from your example here that expects 2.

**Comment by k0da** [comment]
@mallardduck Done

**Review by mallardduck — APPROVED** [review]

**Pull request merged**
